amy/fiscal/tests/test_membership.py

```diff
@@ -1296,3 +1306,33 @@ def test_membership_rollovers(self):
         self.assertEqual(
             last_membership.inhouse_instructor_training_seats_rolled_over, None
         )
+        self.assertEqual(last_membership.rolled_from_membership, self.membership)
+        self.assertEqual(last_membership.rolled_to_membership, None)
+
+    def test_membership_cannot_be_rolled_over_multiple_times(self):
+        # Arrange
+        self.setUpMembership()
+        second_membership = Membership.objects.create(
+            name="Test Membership2",
+            consortium=False,
+            public_status="public",
+            variant="partner",
+            agreement_start="2020-03-01",
+            agreement_end="2021-03-01",
+            contribution_type="financial",
+            workshops_without_admin_fee_per_agreement=10,
+            public_instructor_training_seats=12,
+            additional_public_instructor_training_seats=0,
+            inhouse_instructor_training_seats=9,
+            additional_inhouse_instructor_training_seats=0,
+        )
+        self.membership.rolled_to_membership = second_membership
+        self.membership.save()
+
+        # Act
+        response = self.client.get(
+            reverse("membership_create_roll_over", args=[self.membership.pk])
```

Comment:
On an unrelated note: Why is create membership rollover a get rather than a post?

Reply:
It's a form, it supports both - GET and POST.